Assets/Scripts/Config/ConfigLoader.cs

```diff
@@ -38,46 +34,52 @@ public static string LoadFromStreamingAssets(string relativePath) {
   }
 
   private static T LoadProtobufConfig<T>(
-      string relativePath, Func<string, IntPtr, int, IntPtr, Plugin.StatusCode> loadFunction)
+      string relativePath, Func<string, IntPtr, Plugin.StatusCode> serializedLengthFunction,
+      Func<string, IntPtr, int, IntPtr, Plugin.StatusCode> loadFunction)
       where T : Google.Protobuf.IMessage<T>, new() {
     string streamingAssetsPath = GetStreamingAssetsFilePath(relativePath);
 
-    byte[] buffer = null;
-    int bufferSize = InitialMaxProtobufSerializedLength;
+    // Determine the length of the serialized Protobuf message.
     int serializedLength = 0;
     Plugin.StatusCode status = Plugin.StatusCode.StatusOk;
-    while (true) {
-      buffer = new byte[bufferSize];
-      unsafe {
-        int* serializedLengthPtr = &serializedLength;
-        fixed(void* bufferPtr = buffer) {
-          status = loadFunction(streamingAssetsPath, (IntPtr)bufferPtr, bufferSize,
-                                (IntPtr)serializedLengthPtr);
-        }
-      }
-      if (status == Plugin.StatusCode.StatusOk) {
-        return new Google.Protobuf.MessageParser<T>(() => new T())
-            .ParseFrom(buffer, 0, serializedLength);
-      }
-      if (status == Plugin.StatusCode.StatusFailedPrecondition) {
-        // Double the maximum serialized length and try again.
-        bufferSize *= 2;
-        continue;
+    unsafe {
+      int* serializedLengthPtr = &serializedLength;
+      status = serializedLengthFunction(streamingAssetsPath, (IntPtr)serializedLengthPtr);
+    }
+    if (status != Plugin.StatusCode.StatusOk) {
+      Debug.Log(
+          $"Failed to get the length of the serialized message from the Protobuf text file {relativePath} with status code {status}.");
+      return default;
+    }
+
+    // Load the Protobuf message to binary format.
+    byte[] buffer = new byte[serializedLength];
+    unsafe {
+      int* serializedLengthPtr = &serializedLength;
+      fixed(void* bufferPtr = buffer) {
+        status = loadFunction(streamingAssetsPath, (IntPtr)bufferPtr, serializedLength,
+                              (IntPtr)serializedLengthPtr);
       }
+    }
+    if (status != Plugin.StatusCode.StatusOk) {
       Debug.Log($"Failed to load the Protobuf text file {relativePath} with status code {status}.");
       return default;
     }
```

Comment:
Kind of a minor issue, but the new flow exits immediately whenever `Protobuf_*_LoadToBinary` reports `StatusFailedPrecondition`. That status still happens if the pbtxt changes between the length probe and the load (e.g., someone Ctrl+S's some changes to the file while Unity is still loading the config). 

Before this PR we would retry with a larger buffer and eventually succeed. now we just log and return null, regressing the file-saving-while-running scenario.

Might consider restoring the retry logic that used to be there (i.e. re-run the size-query and retry the load once more; if fail twice, give up)